src/KIM/kim_interactions.cpp

```diff
@@ -276,17 +280,21 @@ void KimInteractions::KIM_SET_TYPE_PARAMETERS(const std::string &input_line) con
     MPI_Bcast(&n,1,MPI_INT,0,world);
     MPI_Bcast(line,n,MPI_CHAR,0,world);
 
-    ptr = line;
     nocomment = line[0] != '#';
 
     if(nocomment) {
       words = utils::split_words(line);
       if (key == "pair") {
+	ptr=line;
+	species1 = strtok(ptr," \t");
```

Comment:
why do you need to use strtok() here when you have the line already split into "words" from the call to `utils::split_words(line)`, if you need to combine words to a longer string again, you can use `fmt::join()`

Reply:
An alternative would be to do 

`std::string the_rest = std::accumulate(words.begin()+2,words.end(),std::string(""),[](std::string const & A,std::string const & B){return A+" "+B;});`

Reply:
you're right - join is better... I've recommitted this, removed all use of strtok and replaced it with a single join.